packages/core/src/elements/FormElement.ts

```diff
@@ -0,0 +1,27 @@
+import type { SourceConfiguration } from '../SourceConfiguration'
+import type { BaseElementEvents } from '../utils/types'
+
+import { SourceElement } from './SourceElement'
+
+export interface FormEvents extends BaseElementEvents {
+  /**
+   * Fired when a form response is submitted
+   */
+  submitted: { id: string }
+}
+
+export interface FormOptions {
+  /**
+   * Provide a formId (or key) into the frame
+   */
+  formId: string
```

Comment:
To be consistent with how we handle appointment types, I'd just call this `form`